tests/capi/unittest_capi_service_agent_client.cc

```diff
@@ -1018,25 +1014,95 @@ TEST_F (MLServiceAgentTest, model_ml_option_get_01_n)
   EXPECT_EQ (ML_ERROR_NONE, status);
 
   gchar *path;
-  status = ml_option_get (info_h, "path", (void **) &path);
+  status = ml_information_get (info_h, "path", (void **) &path);
   EXPECT_EQ (ML_ERROR_NONE, status);
   EXPECT_STREQ (test_model, path);
 
   gchar *description;
-  status = ml_option_get (info_h, "description", (void **) &description);
+  status = ml_information_get (info_h, "description", (void **) &description);
   EXPECT_EQ (ML_ERROR_NONE, status);
   EXPECT_STREQ ("", description);
 
-  status = ml_option_get (info_h, key, NULL);
+
+  status = ml_information_get (info_h, key, NULL);
   EXPECT_EQ (ML_ERROR_INVALID_PARAMETER, status);
 
-  status = ml_option_get (info_h, key, (void **) &value);
+  status = ml_information_get (info_h, key, (void **) &value);
   EXPECT_EQ (ML_ERROR_INVALID_PARAMETER, status);
 
-  status = ml_option_get (NULL, key, (void **) &value);
+  status = ml_information_get (info_h, NULL, (void **) &value);
   EXPECT_EQ (ML_ERROR_INVALID_PARAMETER, status);
 
-  status = ml_option_destroy (info_h);
+  status = ml_information_get (NULL, key, (void **) &value);
+  EXPECT_EQ (ML_ERROR_INVALID_PARAMETER, status);
+
+  status = ml_information_destroy (info_h);
+  EXPECT_EQ (ML_ERROR_NONE, status);
+
+  status = ml_service_model_delete (model_name, 0U);
+  EXPECT_EQ (ML_ERROR_NONE, status);
+
+  g_free (test_model);
+}
+
+/**
+ * @brief Test ml_information_list with invalid param.
+ */
+TEST_F (MLServiceAgentTest, model_ml_information_list_00_n)
+{
+  int status;
+
+  const gchar *model_name = "some_model_name";
+  guint version;
+  const gchar *root_path = g_getenv ("MLAPI_SOURCE_ROOT_PATH");
+
+  /* ml_service_model_register() requires absolute path to model, ignore this case. */
+  if (root_path == NULL)
+    return;
+
+  gchar *test_model = g_build_filename (root_path, "tests", "test_models",
+      "models", "mobilenet_v1_1.0_224_quant.tflite", NULL);
+  ASSERT_TRUE (g_file_test (test_model, G_FILE_TEST_EXISTS));
+
+  status = ml_service_model_delete (model_name, 0U);
+  EXPECT_TRUE (status == ML_ERROR_NONE || status == ML_ERROR_INVALID_PARAMETER);
```

Comment:
I'm sorry, is there a case of status == ML_ERROR_NONE in this TC?

Reply:
It is for delete all models with the key. It can be.

Reply:
Thank you.